test/hotspot/jtreg/serviceability/sa/UniqueVtableTest.java

```diff
@@ -91,42 +78,50 @@ private void detach() {
     }
 
     private void runTest() throws Throwable {
-        Map<Address, List<Type>> types = new HashMap<>();
+        Map<Address, List<Type>> vtableToTypesMap = new HashMap<>();
         Iterator<Type> it = agent.getTypeDataBase().getTypes();
-        int dupFound = 0;
+        int dupsFound = 0;
+        // agent.getTypeDataBase() returns HotSpotTypeDataBase (extends BasicTypeDataBase)
+        BasicTypeDataBase typeDB = (BasicTypeDataBase)(agent.getTypeDataBase());
+        int total = 0;
+        int no_vtable = 0;
+        int no_vtable_with_super = 0;
         while (it.hasNext()) {
+            total++;
             Type t = it.next();
-            Address vtable = (Address) vtblForType.invoke(agent.getTypeDataBase(), t);
-
+            Address vtable = typeDB.vtblForType(t);
             if (vtable != null) {
-                List<Type> typeList = types.get(vtable);
+                no_vtable++;
```

Comment:
`no_vtable` is actually tracking the number of Types with a vtable.

Reply:
Changed reported stats